TheJumpCloud/support
# CUT-4416-Remove-API-Key-Validation

## Issues
* [CUT-4416](https://jumpcloud.atlassian.net/browse/CUT-4416) - CUT-4416-Remove-API-Key-Validation

## What does this solve?

- Remove 40 character API key validation
- Adds the ability to set, get "Notes" on JCPolicy Functions
- Radius Pester Tests are now self-contained (with the addition of one helper function), we kept getting conflicts so the Get, Set, New tests should all create the radius servers necessary for the individual test files instead of defining them globally.

## Is there anything particularly tricky?
n/a
## How should this be tested?
1. Run this branch module and command: `Connect-JCOnline  -JumpCloudApiKey "1234567abc"`
2. Command should run but give API key error: 
<img width="1356" alt="image" src="https://github.com/user-attachments/assets/a6a362d7-3b36-46d7-bcb4-6582fa60ddd7">


JCPolicy features are already written into the pester tests, take a look at those to see the expected behavior for the JCPolicy work

## Screenshots


[CUT-4416]: https://jumpcloud.atlassian.net/browse/CUT-4416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
